interface/modules/zend_modules/module/Carecoordination/view/carecoordination/carecoordination/upload.phtml

```diff
@@ -432,7 +432,7 @@ echo $this->headScript()->prependFile($this->basePath() . '/js/autosuggest/autos
                                     <span><strong style="color: red;"><?php echo $this->escapeHtml($row['matched_patient']); ?></strong></span>
                                 </td>
                                 <td>
-                                    <a href="#" onclick="deleteAuditRecord(<?php echo js_escape($row['amid']); ?>,'<?php echo text($row['pat_name']); ?>')"><strong style="color: red;"><?php echo $this->escapeHtml($row['dupl_patient'] ?? ''); ?></strong></a>
+                                    <a href="#" onclick="deleteAuditRecord(<?php echo intval($row['amid']); ?>,'<?php echo text($row['pat_name']); ?>')"><strong style="color: red;"><?php echo $this->escapeHtml($row['dupl_patient'] ?? ''); ?></strong></a>
```

Comment:
That is really odd on the json_escape. Generally json_escape will leave numbers as they are without any other characters (can try it here http://php.fnlist.com/php/json_encode) (does the $row['amid'] have any leading/trailing spaces in it? If this is possible then running it through trim() prior to js_escape should work).

Reply:
We tried doing the trim.  Got the same behavior.

Here is our comment which we stashed:
+    // TODO: we need to investigate why an integer on linux is coming back with a space
+    // example json_encode(12) => " 12" which breaks in some context.
+    // deleteAuditRecord(<?php echo js_escape($row['amid']); => deleteAuditRecord(" 12");

Reply:
correct escaping for this leading a tag should be (i added the trim in addition to correctly escaping pat_name):
```
<a href="#" onclick="deleteAuditRecord(<?php echo attr_js(trim($row['amid'])); ?>,<?php echo attr_js($row['pat_name']); ?>)">
```

Reply:
> trim($row['amid']

$row['amid'] is an int. trim is not necessary. So then attr_js will not quote $row['amid'] value making this value unnecessary to escape at all. It will always be int.
Regardless, escaping with attr_js should work fine with the way I escaped currently is acceptable where the main issue is why does json_encode cause a pre space on a linux box. Is this a one off or something to do with php version or complied differently for windows. It'd be comforting if you could explain Brady. This is just weird...

Reply:
To clarify it sounds like attr_js will work, so can go with that for the 2 parameters. My guess here is that the double quotes being created by json_encode are messing things up since the onclick is also separated by double quote. Which attr_js() should solve. Just a guess.

Reply:
Yep what you have is fine and should solve issue. It's was disconcerting to us is all.

Reply:
could potential comfirm above hypothesis to see what happens with following:
(perhaps the original was a integer that was treated as a string so brought in double quotes)
```
<a href="#" onclick="deleteAuditRecord(<?php echo js_escape($row['amid']); ?>,'<?php echo text($row['pat_name']); ?>')">
```
versus (force int to avoid the double quotes)
```
<a href="#" onclick="deleteAuditRecord(<?php echo js_escape(intval($row['amid'])); ?>,'<?php echo text($row['pat_name']); ?>')">
```

Reply:
Yes it was intended to get quoted which is why I used js_escape(). Works fine on windows and breaks on Linux by adding a space. I believe this is inherent with json_encode and how it deals with numeric. I learned a lesson here.

Reply:
played around with this a little to try to explain this. here is a dummy php script:
```
<?php
$s = "123";
$i = 123;

echo json_encode($s) . "<br>\n";
echo json_encode($i) . "<br><br>\n";
?>

<?php //js_escape test on string and int (both numbers) ?>
<a href="#" onclick="alert(<?php echo json_encode($s); ?>)">string</a><br>
<a href="#" onclick="alert(<?php echo json_encode(intval($i)); ?>)">string to int</a><br>
<a href="#" onclick="alert(<?php echo json_encode($i); ?>)">int</a><br><br>

<?php // attr_js test on string and int (both numbers) ?>
<a href="#" onclick="alert(<?php echo htmlspecialchars(json_encode($s), ENT_QUOTES); ?>)">string</a><br>
<a href="#" onclick="alert(<?php echo htmlspecialchars(json_encode(intval($i)), ENT_QUOTES); ?>)">string to int</a><br>
<a href="#" onclick="alert(<?php echo htmlspecialchars(json_encode($i), ENT_QUOTES); ?>)">int</a><br><br>
```

screenshot of the screen which appears fine although will get a javascript error if try to click that first string link (all the other links work fine):
![image](https://user-images.githubusercontent.com/278968/168959697-993735d6-660d-46a8-869f-206c83edabf5.png)

screenshot of the source which shows issue because of the nested doublequotes:
![image](https://user-images.githubusercontent.com/278968/168959850-462928d2-8917-451b-a98e-6ca428e200ce.png)

Reply:
Thanks Brady. I'll stick to being careful with integers for now on!